xapian-applications/omega/omindexcheck.cc

```diff
@@ -0,0 +1,142 @@
+/** @file omindexcheck.cc
+ * @brief Auxiliary program of omindextest
+ */
+/* Copyright (C) 2019 Bruno Baruffaldi
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301
+ * USA
+ */
+#include <config.h>
+#include "worker.h"
+
+#include <xapian.h>
+#include <iostream>
+#include <string>
+#include <unordered_map>
+#include <cstdlib>
+
+#include "gnu_getopt.h"
+#include "hashterm.h"
+#include "common/stringutils.h"
+
+using namespace std;
+
+typedef vector<string> testcase;
+
+unordered_map<string, testcase> tests;
```

Comment:
`tests` is only used in this file, so it's better to declare it as `static` - that potentially allows the compiler to optimise better as it then knows that it can't be modified by external functions (unless a pointer or reference to tests is passed to such a function, etc).  It also means fewer external symbols which helps to speed up linking.

Reply:
Okay, I am going to add `static` to `test`